Tests/Unit/Repositories/UserRepositoryTest.php

```diff
@@ -1,4 +1,62 @@
 <?php
 
+use App\Domain\User;
+use Doctrine\ORM\EntityRepository;
+use PHPUnit\Framework\TestCase;
+
+class UserTest extends TestCase
+{
+    /**
+     * @var username|string
+     */
+    private $username;
+    /**
+     * @var email|string
+     */
+    private $email;
+    /**
+     * @var password|string
+     */
+    private $password;
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->user  = $this->createMock(User::class);
+        
+    }
+
+    private function createSut()
+    {
+        return new User(
+            $this->username,
+            $this->email,
+            $this->password
+        );
+    }
+    public function testUser(string $username,string $email,string $password)
+    {
+    
+        $email = $email('test@test.es');
+        $username = $username('test');
+        $password = $password('kjhjj');
+
+        $userRepository = $this->createMock(EntityRepository::class);
+        $userRepository->expects($this->any())
+            ->method('findUserByUsernameOrEmail')
+            ->willReturn($userRepository);
+
+    
+        $objectManager = $this->createMock(ObjectManager::class);
+    
+        $objectManager->expects($this->any())
+            ->method('getRepository')
+            ->willReturn($userRepository);
+
+        $findUserByUsernameOrEmail = new findUserByUsernameOrEmail($objectManager);
```

Comment:
Esto está erróneo, estás haciendo un `new` de una clase que se llama `findUserByUsernameOrEmail` y si te fijas esa clase no exsite. Lo que tenemos que hacer es llamar al `createSut` y de ahí llamar al método `findUserByUsernameOrEmail`, quedaría así `$this->createSut()->findUserByUsernameOrEmail('username', 'email');`, pero para que esto funcione antes has de cambiar la función `createSut`